smack-core/src/main/java/org/jivesoftware/smack/c2s/ModularXmppClientToServerConnection.java

```diff
@@ -1086,4 +1093,8 @@ public void appendStatsTo(ExtendedAppendable appendable) throws IOException {
         }
 
     }
+
+    public interface StreamOpenFactory {
```

Comment:
This should be in its own file within o.j.s.c2s

I also wonder: Don't we need that also for the stream close? Hence I wonder if we should rename this to `StreamOpenAndCloseFactory` (or something?).